src/HttpContext.h

```
@@ -207,9 +206,17 @@ struct HttpContext {
                 /* Continue parsing */
                 return s;
 
-            }, [httpResponseData](void *user, std::string_view data, bool fin) -> void * {
+            }, [httpResponseData, httpContextData](void *user, std::string_view data, bool fin) -> void * {
                 /* We always get an empty chunk even if there is no data */
+
                 if (httpResponseData->inStream) {
+                    if (httpContextData->maxContentLength > 0) {
+                        httpResponseData->total_received_bytes += data.length();
+                        if(httpResponseData->total_received_bytes > httpContextData->maxContentLength) {
+                            us_socket_close(SSL, (us_socket_t *) user, 0, nullptr);
+                            return nullptr;
```

@cirospaciari I am not sure how we should return the error response with the 413 status code here.